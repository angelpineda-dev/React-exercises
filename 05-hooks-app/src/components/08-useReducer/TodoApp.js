import { useEffect, useReducer } from "react";
import { todoReducer } from "./todoReducer";
import "./styles.css";
import TodoList from "./TodoList";
import TodoAdd from "./TodoAdd";

const init = () => {
  return JSON.parse(localStorage.getItem("todos")) || [];
};

const TodoApp = () => {
  const [todos, dispatch] = useReducer(todoReducer, [], init);

  useEffect(() => {
    localStorage.setItem("todos", JSON.stringify(todos));
  }, [todos]);

  const handleDelete = (id) => {
    const action = {
      type: "delete",
      payload: id,
    };

    dispatch(action);
  };

  const handleToggle = (todoId) => {
    dispatch({
      type: "toggle",
      payload: todoId,
    });
  };

  const handleAddTodo = (newTodo) => {
    dispatch({
      type: "add",
      payload: newTodo,
    });
  };

  return (
    <div>
      <h2>Todo App ( {todos.length} ) </h2>

      <div className="row">
        <div className="col-7">
          <TodoList
            todos={todos}
            handleDelete={handleDelete}
            handleToggle={handleToggle}
          />
        </div>
        <div className="col-5 text-center">
          <TodoAdd handleAddTodo={handleAddTodo} />
        </div>
      </div>
    </div>
  );
};

export default TodoApp;
