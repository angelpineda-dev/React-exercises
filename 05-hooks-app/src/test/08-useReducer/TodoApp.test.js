import React from "react";
import TodoApp from "../../components/08-useReducer/TodoApp";
import { mount, shallow } from "enzyme";
//import { demoTodos } from "../fixtures/demoTodos";
//import { act } from "@testing-library/react";

describe("Pruebas en <TodoApp/>", () => {
  test("Debe mostrarse correctamente", () => {
    const wrapper = shallow(<TodoApp />);
  });

  /* 
  // El siguiente test marca un error de memoria ram insuficiente

  test("Debe de agregar un TODO", () => {
    const wrapper = mount(<TodoApp />);

    console.log(wrapper);
  }); */
});
