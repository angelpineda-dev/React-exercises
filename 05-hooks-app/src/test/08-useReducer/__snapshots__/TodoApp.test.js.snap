// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoApp/> Debe mostrarse correctamente 1`] = `
<div>
  <h2>
    Todo App ( 
    0
     ) 
  </h2>
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <TodoList
        handleDelete={[Function]}
        handleToggle={[Function]}
        todos={Array []}
      />
    </div>
    <div
      className="col-5 text-center"
    >
      <TodoAdd
        handleAddTodo={[Function]}
      />
    </div>
  </div>
</div>
`;
